prism/src/main/java/org/apache/falcon/service/BacklogMetricEmitterService.java

```diff
@@ -0,0 +1,355 @@
+                1, recheckInterval, TimeUnit.MILLISECONDS);
+    }
+
+    private void initInstances() throws FalconException {
+        LOG.info("Initializing backlog instances from state store");
+        Map<Entity, List<MetricInfo>> backlogInstances = backlogMetricStore.getAllInstances();
+        if (backlogInstances != null && !backlogInstances.isEmpty()) {
+            for (Map.Entry<Entity, List<MetricInfo>> entry : backlogInstances.entrySet()) {
+                List<MetricInfo> metricsInDB = entry.getValue();
+                List<MetricInfo> metricInfoList = Collections.synchronizedList(metricsInDB);
+                entityBacklogs.put(entry.getKey(), metricInfoList);
+                LOG.debug("Backlog of entity " + entry.getKey().getName() + " for instances " + metricInfoList);
+            }
+        }
+    }
+
+    @Override
+    public void destroy() throws FalconException {
+        scheduledThreadPoolExecutor1.shutdown();
+        scheduledThreadPoolExecutor2.shutdown();
+    }
+
+    @Override
+    public synchronized void onSuccess(WorkflowExecutionContext context) throws FalconException {
+        Entity entity = EntityUtil.getEntity(context.getEntityType(), context.getEntityName());
+        if (entity.getEntityType() != EntityType.PROCESS) {
+            return;
+        }
+        if (entityBacklogs.containsKey(entity)) {
+            List<MetricInfo> metrics = entityBacklogs.get(entity);
+            synchronized (metrics) {
+                Date date = SchemaHelper.parseDateUTC(context.getNominalTimeAsISO8601());
+                backlogMetricStore.deleteMetricInstance(entity.getName(), context.getClusterName(),
+                        date, entity.getEntityType());
+                metrics.remove(new MetricInfo(DATE_FORMAT.get().format(date), context.getClusterName()));
+                if (metrics.isEmpty()) {
+                    entityBacklogs.remove(entity);
+                }
+            }
+        }
+    }
+
+    @Override
+    public void onFailure(WorkflowExecutionContext context) throws FalconException {
+        // Do Nothing
+    }
+
+    @Override
+    public void onStart(WorkflowExecutionContext context) throws FalconException {
+        // Do Nothing
+    }
+
+    @Override
+    public void onSuspend(WorkflowExecutionContext context) throws FalconException {
+        // Do Nothing
+    }
+
+    @Override
+    public void onWait(WorkflowExecutionContext context) throws FalconException {
+        // Do Nothing
+    }
+
+    /**
+     * Service which executes backlog evaluation and publishing metrics to Graphite parallel for entities.
+     */
+    public static class BacklogMetricEmitter implements Runnable {
+        private ThreadPoolExecutor executor;
+
+        @Override
+        public void run() {
+            LOG.debug("BacklogMetricEmitter running for entities");
+            executor = new ScheduledThreadPoolExecutor(10);
```

Comment:
Lets keep the thread number higher around 100 as the for loop will make only use as per requirement.